### PR TITLE
v: Improve C var args interop

### PR DESCRIPTION
**Pull request by felipensp**
This PR:
- Introduce simple `...` syntax for function declaration (to use C var args)
- Makes possible to declare `C.va_list{}` without initialization.

```v
@[typedef]
struct C.va_list {}
fn C.va_start(voidptr, voidptr)
fn C.va_end(voidptr)

fn C.vfprintf(&C.FILE, &char, C.va_list) int

fn t2(fmt voidptr, ...) {
	ap := C.va_list{}
	C.va_start(ap, fmt)
	C.vfprintf(C.stderr, fmt, ap)
	C.va_end(ap)
}

fn main() {
	a := 1
	b := 2.5
	t2(c'%s : %d : %.2f\n', c'foo', a, b)
}
```

Outputs `foo : 1 : 2.50`

### CODE REVIEW

File: vlib/v/gen/c/cheaders.v [hunk 1]
```diff
@@ -342,7 +342,7 @@ const c_common_macros = '
 	#define TCCSKIP(x)
 	// #include <byteswap.h>
 	#ifndef _WIN32
-		int tcc_backtrace(const char *fmt, ...);
+		int tcc_backtrace(char *fmt, ...);
```
Comment (spytheman) [inline vlib/v/gen/c/cheaders.v:345]: Why is that needed?

Comment (spytheman) [reply vlib/v/gen/c/cheaders.v:345]: Did it change in latest tcc?

Comment (spytheman) [reply vlib/v/gen/c/cheaders.v:345]: Hm, it seems to be still declared with const there:
```c
int tcc_backtrace(const char *fmt, ...);
```

https://repo.or.cz/tinycc.git/blob/HEAD:/lib/bcheck.c#l448

Comment (felipensp) [reply vlib/v/gen/c/cheaders.v:345]: Yeah, this will not be needed to change. Reverting this part.

**Comment by spytheman** [comment]
Excellent work, but it misses a test, and it does not fix #20636 yet (which uses also `va_arg`, and `v translate` does not handle well neither va_start, va_end nor va_arg ... va_arg also accepts a *type name* as its second argument, which can not be modeled currently in V, except perhaps as a generic).

I suggest adding a test .vv file, in `vlib/v/gen/c/testdata/`, that includes a .c file containing the definition of a variadic function printing its arguments, then declaring and calling it in the .vv file, and having an .out file with the printed output.

Then `./v vlib/v/gen/c/coutput_test.v` will pick up the file and check that it continues to work locally and on the CIs.

**Comment by spytheman** [comment]
The current example can also be copy/pasted as a _keep.vv file in `vlib/v/fmt/tests/`, to ensure that `vfmt` will continue to handle it.
